File: tests/rolling_alpha_test.py
import numpy as np

from taflow import RollingAlpha


def test_rolling_alpha_lifecycle_and_alignment():
    state = RollingAlpha(
        np.array([], dtype=float), np.array([], dtype=float), timeperiod=2
    )
    state.extend([1.0, 2.0], [2.0, 4.0])
    np.testing.assert_allclose(state.compute(), [np.nan, 0.0], equal_nan=True)
    assert len(state) == 2
    state.reset()
    assert state.value is None

File: crates/taflow-core/src/indicators/rolling_alpha_test.rs
use super::rolling_alpha::RollingAlpha;

#[test]
fn warmup_and_reset_are_consistent() {
    let mut state = RollingAlpha::new(2).unwrap();
    assert_eq!(state.append(1.0, 2.0), None);
    assert!(state.append(2.0, 4.0).is_some());
    state.reset();
    assert_eq!(state.value(), None);
}
